packages/server/src/config.ts

```diff
@@ -66,13 +72,33 @@ function databaseHostFromEnv(runningInDocker: boolean, env: EnvVariables): strin
 
 function databaseConfigFromEnv(runningInDocker: boolean, env: EnvVariables): DatabaseConfig {
 	if (env.DB_CLIENT === 'pg') {
+		let databaseConfig: DatabaseConfig = {
+			client: DatabaseConfigClient.PostgreSQL
+		};
+
+		if (env.POSTGRES_CONNECTION_STRING) {
+			databaseConfig.connectionString = env.POSTGRES_CONNECTION_STRING;
+		} else {
+			databaseConfig.name = env.POSTGRES_DATABASE || 'joplin';
+			databaseConfig.user = env.POSTGRES_USER || 'joplin';
+			databaseConfig.password = env.POSTGRES_PASSWORD || 'joplin';
+			databaseConfig.port = env.POSTGRES_PORT ? Number(env.POSTGRES_PORT) : 5432;
+			databaseConfig.host = databaseHostFromEnv(runningInDocker, env) || 'localhost';
+		}
+
+		if(env.POSTGRES_REJECT_UNAUTHORIZED)
+			databaseConfig.rejectUnauthorized = Number(env.POSTGRES_REJECT_UNAUTHORIZED) === 1;
+
+		if(env.POSTGRES_SSL_CA_FILEPATH)
+			databaseConfig.sslCaFilePath = env.POSTGRES_SSL_CA_FILEPATH;
+
+		if(env.POSTGRES_SSL_CERT_FILEPATH && env.POSTGRES_SSL_CERT_KEY_FILEPATH) {
+			databaseConfig.sslCertFilePath = env.POSTGRES_SSL_CERT_FILEPATH;
+			databaseConfig.sslCertKeyFilePath = env.POSTGRES_SSL_CERT_KEY_FILEPATH;
+		}
```

Comment:
Please keep the earlier structure - it should be just one return statement, and each property is loaded from the environment. Doing it that way ensures all properties and are assigned good defaults. For example:

```typescript
connectionString: env.POSTGRES_CONNECTION_STRING || '',
databaseConfig.sslCert = env.POSTGRES_SSL_CERT_FILE ? await fs.readFile(env.POSTGRES_SSL_CERT_FILE) : null,
// etc.
```

Reply:
the main reason I have done it this way was not to set these values at all if they were not needed. Happy to change this up though.

Reply:
a few thoughts here is that if there is a connection string supplied there should not be other connection props? will this get confusing when debugging connection issues? as this is logged out when the server starts. Also in the case of `sslCerts` and `sslCertKey` one should not exist with out the other

Reply:
The role of this function is simply to build up the config object, which is then used in other parts of the code, and it's in this other parts that there should be conditionals to decide what should be used or not.

If you put business logic here you'll have to duplicate it elsewhere, and it's indeed what you've done since you've duplicated all this logic in the db.ts file. That's a small thing for now, but as the config grows bigger we don't want to have to keep its logic in sync with everywhere else it might be used.

So here we process the env variables, all of them, and assign them to config properties.